core/src/services/hdfs_native/lister.rs

```diff
@@ -15,28 +15,27 @@
 // specific language governing permissions and limitations
 // under the License.
 
-use std::sync::Arc;
+use std::collections::VecDeque;
 
 use crate::raw::oio;
-use crate::raw::oio::Entry;
-use crate::*;
+use crate::Result;
 
 pub struct HdfsNativeLister {
-    _path: String,
-    _client: Arc<hdfs_native::Client>,
+    entries: VecDeque<oio::Entry>,
 }
 
 impl HdfsNativeLister {
-    pub fn new(path: String, client: Arc<hdfs_native::Client>) -> Self {
-        HdfsNativeLister {
-            _path: path,
-            _client: client,
-        }
+    pub fn new(entries: VecDeque<oio::Entry>) -> Self {
+        HdfsNativeLister { entries }
     }
 }
 
 impl oio::List for HdfsNativeLister {
-    async fn next(&mut self) -> Result<Option<Entry>> {
-        todo!()
+    async fn next(&mut self) -> Result<Option<oio::Entry>> {
```

Comment:
We should call the `list_status` API here instead, so we don't have to store all entries in memory before the user calls `next`.

In the future, we could also support using `LIST_BATCH` from HDFS, allowing users to call `list_status` multiple times to fetch all entries.

Reply:
Makes sense